### PR TITLE
Enhancement/terms lookup fixes

### PR DESCRIPTION
**Pull request by javanna**
Relates to https://github.com/elastic/elasticsearch/pull/12042#issuecomment-131024096 .
- Java api: remove support for lookup cache in TermsLookupBuilder: TermsQueryParser doesn't support the cache field anymore, so if it gets set through java api, the subsequent parsing of that query will throw error
- Java api: restore support for minimumShouldMatch and disableCoord in TermsQueryBuilder: TermsQueryParser still parses those values although deprecated. These need to be present in the java api as well to get ready for the query refactoring, where the builders are the intermediate query format that we parse our json queries into. Whatever the parser supports need to be supported by the builder as well.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java [hunk 1]
```diff
@@ -126,6 +130,26 @@ public TermsQueryBuilder execution(String execution) {
...
+        this.minimumShouldMatch = minimumShouldMatch;
+        return this;
+    }
+
+    /**
+     * Disables <tt>Similarity#coord(int,int)</tt> in scoring. Defaults to <tt>false</tt>.
+     * @deprecated use [bool] query instead
+     */
+    @Deprecated
+    public TermsQueryBuilder disableCoord(boolean disableCoord) {
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java]: Thanks for doing this, can you also make the parser use a ParseField that marks it as deprecated?

Comment (javanna) [inline core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java]: good point, done

**Comment by jpountz** [comment]
LGTM

**Comment by javanna** [comment]
Marking this as breaking as the removal of the setter for `lookupCache` from `TermsLookupBuilder` breaks the java api.

**Pull request closed without merging**